packages/turf-boolean-point-on-line/index.ts

```diff
@@ -97,6 +97,25 @@ function isPointOnLineSegment(
   } else if (cross !== 0) {
     return false;
   }
+
+  // Special cases for zero length lines
+  // https://github.com/Turfjs/turf/issues/2750
+  if (Math.abs(dxl) === Math.abs(dyl) && Math.abs(dxl) === 0) {
```

Comment:
Does this math need to be done with some small epsilon for the float comparison instead of `===`?

Reply:
It should be ok without. The new section is guarding against the old (lower) code getting too many `0`s for dxl, etc. They don't seem to behave if they do.

If they get any value above zero, no matter how small, they cope. For example, did some experiments changing the test line string to:

```typescript
lineString([
  [1, 1],
  [1 + Number.EPSILON, 1],
])
```

and all is well.

Reply:
I could try incorporating additional checks into the existing ```return dxl > 0 ? x1 <= x && x <= x2 : x2 <= x && x <= x1;``` type code but keeping them separate seemed neater.